Devnet: new image release

pLabarta opened the pull request.

zacharyfrederick committed:
Merge pull request #6 from gluwa/dev

Build Crunch Testnet Image

pLabarta committed:
Merge pull request #7 from gluwa/devnet

Merge pull request #4 from gluwa/dev

pLabarta committed:
Update testnet.yml

pLabarta committed:
Update testnet.yml

Commit:
removed dynamic validator list functionality

pLabarta committed:
add: testnet validator list

pLabarta committed:
add: env files, devnet stashes & remove from gitignore

pLabarta committed:
rm: config step from testnet & devnet workflows

pLabarta committed:
Merge branch 'testnet' into dev

pLabarta committed:
add: default stash in config

Commit:
updated runtime to fully execute batch payouts

Commit:
removed unused logging messages

Commit:
more cleanup

Commit:
reverted unneeded changes to sleep timing

pLabarta committed:
update: devnet config & stashes

atodorov reviewed (changes requested):
This looks iffy to me. 

- What is the relationship between `dev` and `devnet` branches? 
- Aren't we supposed to use a feature branch and merge into `dev` before opening any changes/PRs against other branches? 
 
It feels like we can make a huge mess if we push commits directly into `dev`

frankli-dev commented:
We are going to create `release/*` branch for release process. Gonna not use `devnet` or `testnet`.

notaturkey committed:
docker images are built on release/* branch (#15)

atodorov committed:
Rename release related workflow to release.yml

atodorov committed:
Copy over create_release.yml from upstream again where their tests are

because previous commits didn't leave this file alone but rather made
lots of changes into it and then subsequently removed it.

atodorov committed:
Rename upstream create_release.yml into ci.yml

will add subsequent changes here to make tests trigger on each PR
however this is a good starting point

atodorov committed:
Enable minimal testing on PRs

- cargo check
- cargo clippy
- cargo fmt
- cargo test

atodorov committed:
Enable pre-commit CI

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

atodorov committed:
Enable MegaLinter

- disable Clippy b/c it will be enabled separately as part of several
  other cargo CI jobs
- disable spell checkers - too many false positives

atodorov committed:
Update README files to satisfy linters

atodorov committed:
Revert "initial" to bring back deleted LICENSE file

This reverts commit ca51756f500b28694d5fafdd3748a27673977819.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

atodorov committed:
Fix issues reported by ShellCheck

- must use double quotes

atodorov committed:
Configure top-level permissions in GHA to read-all

atodorov committed:
Ignore git leaks from upstream commits

atodorov committed:
Improvements to Dockerfile(s)

- ignore
  The instruction 'RUN <package-manager> update' should always be followed by '<package-manager> install' in the same RUN statement.
  See https://avd.aquasec.com/misconfig/ds017

  which is triggered by `rustup update`

atodorov committed:
Enable Dependabot

zacharyfrederick committed:
removed references to the unused matrix.rs module

zacharyfrederick committed:
remove matrix.rs for megalinter

zacharyfrederick committed:
add .jscpd.json configuration file and set threshold to 5

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

zacharyfrederick committed:
updated .jscpd.json config

zacharyfrederick committed:
added a healtcheck to the crunch bot and added instructions to the dockerfile

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

zacharyfrederick committed:
update the correct dockerfiles this time

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

atodorov committed:
Add a git diff as part of MegaLinter's auto-commit output

b/c we need a personal access token here in order for this to work, see
https://github.com/stefanzweifel/git-auto-commit-action/issues/322

atodorov committed:
Reformat

atodorov committed:
Reformat with cargo fmt

atodorov committed:
Stricter checks in CI

atodorov committed:
Remove unused imports

atodorov committed:
Silence warnings for unused variables

for the upstream code just prefix them with underscore to minimize
differences, for creditcoin/ just remove them

atodorov committed:
Remove unnecessary mut

atodorov committed:
Allow 2 unused methods from upstream implementation

atodorov committed:
Add rust-toolchain.toml and use the same rustc version in CI

because fmt and clippy appear to be slightly different from version to
version and we'd like to have consistency regardless of how or where
this is built

atodorov committed:
Reformat again with rustc 1.77.0

atodorov committed:
Apply suggestions from clippy

atodorov committed:
Adjust CRUNCH_CONFIG_FILENAME to point to an existing file

because .env.example has been removed in
https://github.com/gluwa/crunch/commit/ca51756f500b28694d5fafdd3748a27673977819

atodorov committed:
Collect and upload code coverage when executing unit tests

dependabot[bot] committed:
Bump taiki-e/install-action from 2.27.14 to 2.32.7

Bumps [taiki-e/install-action](https://github.com/taiki-e/install-action) from 2.27.14 to 2.32.7.
- [Release notes](https://github.com/taiki-e/install-action/releases)
- [Changelog](https://github.com/taiki-e/install-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/taiki-e/install-action/compare/v2.27.14...v2.32.7)

---
updated-dependencies:
- dependency-name: taiki-e/install-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump azure/login from 1 to 2

Bumps [azure/login](https://github.com/azure/login) from 1 to 2.
- [Release notes](https://github.com/azure/login/releases)
- [Commits](https://github.com/azure/login/compare/v1...v2)

---
updated-dependencies:
- dependency-name: azure/login
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update env_logger requirement from 0.9.3 to 0.11.3

Updates the requirements on [env_logger](https://github.com/rust-cli/env_logger) to permit the latest version.
- [Release notes](https://github.com/rust-cli/env_logger/releases)
- [Changelog](https://github.com/rust-cli/env_logger/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rust-cli/env_logger/compare/v0.9.3...v0.11.3)

---
updated-dependencies:
- dependency-name: env_logger
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update base64 requirement from 0.13.0 to 0.22.0

Updates the requirements on [base64](https://github.com/marshallpierce/rust-base64) to permit the latest version.
- [Changelog](https://github.com/marshallpierce/rust-base64/blob/master/RELEASE-NOTES.md)
- [Commits](https://github.com/marshallpierce/rust-base64/compare/v0.13.0...v0.22.0)

---
updated-dependencies:
- dependency-name: base64
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update async-recursion requirement from 0.3.2 to 1.1.0

Updates the requirements on [async-recursion](https://github.com/dcchut/async-recursion) to permit the latest version.
- [Release notes](https://github.com/dcchut/async-recursion/releases)
- [Commits](https://github.com/dcchut/async-recursion/compare/v0.3.2...v1.1.0)

---
updated-dependencies:
- dependency-name: async-recursion
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update reqwest requirement from 0.11 to 0.12

Updates the requirements on [reqwest](https://github.com/seanmonstar/reqwest) to permit the latest version.
- [Release notes](https://github.com/seanmonstar/reqwest/releases)
- [Changelog](https://github.com/seanmonstar/reqwest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/seanmonstar/reqwest/compare/v0.11.0...v0.12.2)

---
updated-dependencies:
- dependency-name: reqwest
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
